packages/core/src/Stores/client-store.js

```diff
@@ -1224,12 +1223,26 @@ export default class ClientStore extends BaseStore {
      */
     @action.bound
     async init(login_new_user) {
-        const search = window.location.search;
-        const search_params = new URLSearchParams(search);
+        this.initial_search_params = window.location.search;
+        const search_params = new URLSearchParams(this.initial_search_params);
```

Comment:
Here, I think it's better to store the search parameters in a const and remove them immediately. We can pass around the stored search parameters wherever it is required.